lib/_http_outgoing.js

```diff
@@ -407,9 +424,10 @@ Object.defineProperty(OutgoingMessage.prototype, 'headersSent', {
 
 OutgoingMessage.prototype.write = function(chunk, encoding, callback) {
   var self = this;
+  var err;
 
   if (this.finished) {
-    var err = new Error('write after end');
+    err = new Error('write after end');
```

Comment:
I left that return value unchanged. It was that way I before I got there! I swear! ;)

The reason for not throwing at that point is that the user *still* has no reliable (documented!) way of checking the state of the stream beforehand. He would have to wrap every write in try catch...